ParcialProgramacion4/src/Empresa.java

```diff
@@ -517,24 +517,38 @@ public void borrarPuesto() {
 
         Puesto puestoBorrar = this.buscarPuesto(nombrePuesto);
 
-        if (puestoBorrar != null) {
+        if (puestoBorrar == null) {
+            Logger.logError("NO existe puesto de trabajo con este nombre");
+        } else {
             Logger.header("Informacion puesto a eliminar: ");
             puestoBorrar.mostrar();
 
             int cantEmpleados = puestoBorrar.cantEmpleados();
 
-            if (cantEmpleados == 0) {
-                puestos.remove(puestoBorrar);
+            if (cantEmpleados != 0) {
+                Logger.logError("NO se puede eliminar, porque "+ cantEmpleados + " empleados tienen este puesto");
+            } else {
+                //buscar si esta en alguna convocatoria
+                boolean estaEnConvocatorias = puestoEstaEnConvocatorias(puestoBorrar);
 
-                Logger.logSuccess("Puesto de trabajo ELIMINADO");
+                if (estaEnConvocatorias) {
+                    Logger.logError("NO se puede eliminar, porque el puesto de " + nombrePuesto + " esta en convocatorias");
+                } else {
+                    puestos.remove(puestoBorrar);
 
-            } else {
-                Logger.logError("NO se puede eliminar, porque "+ cantEmpleados + " empleados tienen este puesto");
+                    Logger.logSuccess("Puesto de trabajo " + nombrePuesto + " ha sido ELIMINADO");
+                }
             }
+        }
+    }
 
-        } else {
-            Logger.logError("NO existe puesto de trabajo con este nombre");
+    private boolean puestoEstaEnConvocatorias(Puesto puesto) {
```

Comment:
for (int i = 0; i < convocatorias.size(); i++) {
    if (convocatorias.get(i).hasPuesto(puesto)) {
        return true;
    }
}
return false;

Reply:
Lo hizo con while por si corta antes porque una ya lo tiene, no tenga que analizar todas las convocatorias